source/app/util/custom_resource.py

```diff
@@ -145,6 +145,9 @@ def handle_request(self):
         timer = threading.Timer(timeleft, self.fn_timeout)
         timer.start()
 
+        # default error to return if something causes the try to fail (this has been freezing for some reason)
+        self.response["Status"] = "FAILED"
+        self.response["Reason"] = "Unknown Error"
```

Comment:
What problem does this solve specifically, and how?

A comment like this is worrying. We shouldn't be making changes to try random things. If we don't know how it works, we need to find out how and make it work the right way. I understand that is breaking the pattern in this solution, but that needs to change.